src/billing/billing-address-action-creator.ts

```diff
@@ -50,17 +53,24 @@ export default class BillingAddressActionCreator {
                 };
             }
 
-            observer.next(createAction(BillingAddressActionType.ContinueAsGuestRequested));
-
-            this._createOrUpdateBillingAddress(checkout.id, billingAddressRequestBody, options)
-                .then(({ body }) => {
-                    observer.next(createAction(BillingAddressActionType.ContinueAsGuestSucceeded, body));
-                    observer.complete();
-                })
-                .catch(response => {
-                    observer.error(createErrorAction(BillingAddressActionType.ContinueAsGuestFailed, response));
-                });
-        });
+            return merge(
+                concat(
+                    of(createAction(BillingAddressActionType.ContinueAsGuestRequested)),
+                    defer(async () => {
+                        const { body } = await this._createOrUpdateBillingAddress(
+                            checkout.id,
+                            billingAddressRequestBody,
+                            options
+                        );
+
+                        return createAction(BillingAddressActionType.ContinueAsGuestSucceeded, body);
+                    }).pipe(
```

Comment:
I'd probably move this `pipe` to the end of `concat` because your intention is to catch any error that could happen within this `concat` block. Having said that, there's only one observable that could potentially fail - that is the `defer`. So it really makes no difference at the moment. But if that's the case, you can just simplify this by using a regular `try/catch` block within the `defer`. 🍹